[Ruizhi2001] iP

Ruizhi2001 opened the pull request:
# Ruiz! 🥳
> "The brain 🧠 is for thinking, the pencil ✍️ is for remembering." - Random quote I found online
### Ruiz does _**ALL**_ the dirty work for you. It
-  **REMEMBERS** all your tasks 🤓
-  is **EASY** to use 🥱
-  is **INTERACTIVE** 🤖
-  ~~washes your dishes 🍽~~
### You just need to
1.  fork this [repository](https://github.com/Ruizhi2001/ip)
2.  run it
3.  use it through the command line
4. focus on the important things 🤩
### And it's **FREE! 🤑**

## 👩‍💻Features🧑‍💻
- [X] Managing tasks
- [ ] Widgets
- [ ] Reminders

## If you're feeling adventurous 🧗🏻‍♀️, you can customize it even more! 
### Here's a sneak peek 🫣 of the `main` method:
```java
public static void main(String[] args) {
        new Ruiz(filePath).run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Ruizhi2001 committed:
Level 0. Rename, Greet, Exit

Ruizhi2001 committed:
Level-1: Echo

Ruizhi2001 committed:
Level 2. Add, List

Ruizhi2001 committed:
Level 3. Mark as Done

Ruizhi2001 committed:
Level 4. ToDos, Events, Deadlines

Ruizhi2001 committed:
A-TextUiTesting

Ruizhi2001 committed:
Level 5. Handle Errors

Ruizhi2001 committed:
Level 5. Handle Errors

Ruizhi2001 committed:
Level 6. Delete
Error handling added for delete, mark and unmark

Ruizhi2001 committed:
Some formatting

Ruizhi2001 committed:
Minor edits

Ruizhi2001 committed:
Added javadocs and fixed minor bugs

Ruizhi2001 committed:
More robust testing

Ruizhi2001 committed:
Add ability to store tasks in hard disk of user

Ruizhi2001 committed:
Merge branch 'branch-Level-7'

Ruizhi2001 committed:
Add ability to accept dates in a format such as yyyy-mm-dd format (e.g., 2019-10-15) and print in a different format such as MMM dd yyyy.

Ruizhi2001 committed:
Merge branch 'branch-Level-8'

Ruizhi2001 committed:
Fix storage bugs

Ruizhi2001 committed:
add more OOP through the use of extra classes "Storage", "Command", "Parser" + " TaskList" and "Ui"

Ruizhi2001 committed:
Merge branch 'A-MoreOOP'

Ruizhi2001 committed:
Add the use of packages for each class

Ruizhi2001 committed:
Merge branch 'A-Packages'

Ruizhi2001 committed:
Set up gradle to build and run Duke using gradle.

Ruizhi2001 committed:
Add gradle to the master branch

Ruizhi2001 committed:
Add testcases to JUnit for the Parser and ToDos

Ruizhi2001 committed:
Merge branch 'A-JUnit'

Ruizhi2001 committed:
Change the message that prints when there is no pre-existing task file saved.

Ruizhi2001 committed:
Add JavaDocs to most non-trivial methods.

Ruizhi2001 committed:
Merge branch 'A-JavaDoc'

Ruizhi2001 committed:
Add Level-9 ability to search using keyword

Ruizhi2001 committed:
Format some code

Ruizhi2001 committed:
Reformat code to align with coding standards

Ruizhi2001 committed:
Merge branch 'A-CodingStandard'

# Conflicts:
#	src/test/java/ruiz/ParserTest.java

Ruizhi2001 committed:
Merge branch 'Level-9'

Ruizhi2001 committed:
Clean up code and add javadocs to more methods

Ruizhi2001 committed:
Set up stylecheck and edit the files to comply with the stylecheck rules.

chiayunrong commented on src/main/java/ruiz/Parser.java:
```diff
@@ -0,0 +1,159 @@
+            throw new BotException("OOPS!!! The description the event is incomplete."
+                    + "it should be in the format 'event *** /from *** /to ***'");
+        }
+        return input.split(" ", 2)[1].split(" /from ")[1].split(" /")[0];
```
Its not clear immediately what the chain of splits is supposed to achieve to me. Maybe each split can be split off into its own descriptive function or something similar to improve readability?

chiayunrong commented on src/test/java/ruiz/ToDosTest.java:
```diff
@@ -0,0 +1,26 @@
+
+public class ToDosTest {
+    @Test
+    public void saveToString_success() {
```
I'm not really sure what this test is supposed to test, perhaps it could be changed to something like testMarkTodoSuccess?

chiayunrong commented on src/test/java/ruiz/ToDosTest.java:
```diff
@@ -0,0 +1,26 @@
+    @Test
+    public void saveToString_success() {
+        ToDos test = new ToDos("eat");
+        assertEquals("T | 0 | eat", test.saveTaskString());
```
Each unit test should only test one thing. However this test seems to be testing 2 things: 
1. whether the todo is created appropriately
2. whether the todo is able to be marked properly 

Do consider splitting the tests into different test methods.

chiayunrong reviewed (commented):
Great work overall! Only a few minor comments.

Ruizhi2001 committed:
Add GUI

Ruizhi2001 committed:
Merge branch 'branch-A-Level-10'

joeng03 commented on src/main/java/ruiz/task/TaskList.java:
```diff
@@ -0,0 +1,186 @@
+     * @throws BotException if the input is not a valid one.
+     */
+    public String markTask(String input) throws BotException {
+        if (input.split(" ").length < 2) {
+            throw new BotException("OOPS!!! The index of a task cannot be empty.");
+        }
+        try {
+            int taskIndex = Integer.parseInt(input.split(" ")[1]) - 1;
+        } catch (NumberFormatException e) {
+            throw new BotException("OOPS!!! The index of a task has to be an integer.");
+        }
+        int taskIndex = Integer.parseInt(input.split(" ")[1]) - 1;
```
Might want to abstract out the input parsing and error handling, to supply 'clean' indices as integers to the methods in TaskList. Note that this is not a coding standard violation.

joeng03 commented on src/main/java/ruiz/task/TaskList.java:
```diff
@@ -0,0 +1,186 @@
+     * @throws BotException if the input is not a valid one.
+     */
+    public String unmarkTask(String input) throws BotException {
+        if (input.split(" ").length < 2) {
+            throw new BotException("OOPS!!! The index of a task cannot be empty.");
+        }
+        try {
+            int taskIndex = Integer.parseInt(input.split(" ")[1]) - 1;
+        } catch (NumberFormatException e) {
+            throw new BotException("OOPS!!! The index of a task has to be an integer.");
+        }
+        int taskIndex = Integer.parseInt(input.split(" ")[1]) - 1;
```
Same as above.

joeng03 commented on src/main/java/ruiz/task/ToDos.java:
Would it be more intuitive if ToDos, Deadlines, and Events classes were named in their singular form instead, as their instances are created individuals (i.e. new Event() means creating one new event)?

joeng03 reviewed (commented):
Overall great job! Solid OOP and tests. Coding standards were also well adhered to. Just a few nits

angkyakdifp commented on src/main/java/ruiz/Ui.java:
```diff
@@ -0,0 +1,121 @@
+public class Ui {
+    /**
+     * This method prints the greeting message of the bot.
+     */
```
Is this method prints out the greeting message or just returning a string?

angkyakdifp commented on src/main/java/ruiz/Ui.java:
```diff
@@ -0,0 +1,121 @@
+     */
+    public String wrongFormat() {
+        return "Please input your date and time in the correct format: yyyy-MM-dd HHmm";
+    }
```
Same as above

angkyakdifp commented on src/main/java/ruiz/Ui.java:
```diff
@@ -0,0 +1,121 @@
+    /**
+     * Prints an acknowledgement message that the task has been unmarked.
+     *
+     * @param task task that is marked.
```
Does the task has to be marked when we call this method?

angkyakdifp replied:
Does this method also print the message?

angkyakdifp commented on src/main/java/ruiz/task/TaskList.java:
```diff
@@ -0,0 +1,186 @@
+        return this.taskList.size();
+    }
+
+    /**
+     * This method marks the given task in the input.
+     *
+     * @param input The string that consists of the keyWord "mark" and task index being input by the user.
+     * @throws BotException if the input is not a valid one.
+     */
```
Since this is not a void method, what does this method returns?

angkyakdifp commented on src/main/java/ruiz/task/TaskList.java:
```diff
@@ -0,0 +1,186 @@
+        }
+    }
+
+    /**
+     * This method deletes the specified task from its index from the list of tasks.
+     *
+     * @param input The string that contains the keyWord "delete" and the index of the task.
+     * @throws BotException if the input of the user is not a valid one.
+     */
```
Same as above

angkyakdifp commented on src/main/java/ruiz/task/TaskList.java:
```diff
@@ -0,0 +1,186 @@
+        }
+    }
+
+    /**
+     * This method unmarks the specified task in the string input.
+     *
+     * @param input The string that consists of the keyWord "unmark" and task index being input by the user.
+     * @throws BotException if the input is not a valid one.
+     */
```
Same as above

angkyakdifp commented on src/main/java/ruiz/task/TaskList.java:
```diff
@@ -0,0 +1,186 @@
+        }
+    }
+
+    /**
+     * This method creates a todo and adds it to the list of tasks.
+     *
+     * @param input contains the keyWord "todo" and the description of the todo.
+     * @throws BotException if the input is not in the format of a valid todo.
+     */
```
Same as above

angkyakdifp commented on src/main/java/ruiz/task/TaskList.java:
```diff
@@ -0,0 +1,186 @@
+        return ui.addedNewTaskMsg(temp, this.getTaskListSize());
+    }
+
+    /**
+     * This method creates a deadline and adds it to the list of tasks.
+     *
+     * @param input contains the keyWord "deadline" and the description of the deadline
+     *              with the time it needs to be completed by.
+     * @throws BotException if the input is not in the valid format of a deadline.
+     */
```
Same as above

angkyakdifp commented on src/main/java/ruiz/task/TaskList.java:
```diff
@@ -0,0 +1,186 @@
+        return ui.addedNewTaskMsg(temp, this.getTaskListSize());
+    }
+
+    /**
+     * This method creates an event and adds it to the list of tasks.
+     *
+     * @param input contains the command "event" and the description of the event
+     *              with the time it takes place from and ends by.
+     * @throws BotException if the input is not in the form a valid event.
+     */
```
Same as above

angkyakdifp commented on src/main/java/ruiz/task/TaskList.java:
```diff
@@ -0,0 +1,186 @@
+        return ui.addedNewTaskMsg(temp, this.getTaskListSize());
+    }
+
+    /**
+     * This method finds tasks that contain the keyword passed in.
+     *
+     * @param input contains the command "find" and the "keyword"
+     * @throws BotException if the input is not in the form a valid "find".
+     */
```
Same as above

angkyakdifp commented on src/main/java/ruiz/task/TaskList.java:
```diff
@@ -0,0 +1,186 @@
+     *
+     * @return size of taskList.
+     */
+    public int getTaskListSize() {
+        return this.taskList.size();
+    }
```
Great use of CamelCase

angkyakdifp reviewed (commented):
LGTM. Just need to fix minor issues

Ruizhi2001 committed:
TaskList Class: extract parsing mechanisms

TaskList still handles some parsing behaviours (e.g for the extraction
of task id)

The parsing behaviour is extracted to a new method in the Parser class.

Creating a new method allows us to consolidate all the parsing
behaviours.

Ruizhi2001 committed:
Add assertions : add assertions throughout the project

Adding assertions can reduce the number of bugs in the project.

Assertion statements are added in most classes to make sure the methods
work as intended.

It is done such as to ensure a majority of the project is covered by
assertions.

Ruizhi2001 committed:
Improved on method field names.

There are several cases when method and field names are not specific
enough.

Fixing their names improves code quality.

Ruizhi2001 committed:
Merge pull request #1 from Ruizhi2001/branch-A-Assertions

Branch a assertions

Ruizhi2001 committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/ruiz/Ruiz.java
#	src/main/java/ruiz/task/Deadline.java
#	src/main/java/ruiz/task/Event.java
#	src/main/java/ruiz/task/TaskList.java
#	src/main/java/ruiz/ui/Ui.java
#	src/main/java/ruiz/utils/Storage.java
#	src/test/java/ruiz/ToDoTest.java

Ruizhi2001 committed:
Add use of streams in project.

Ruizhi2001 committed:
Merge branch 'master' into branch-A-Streams

Ruizhi2001 committed:
Merge pull request #3 from Ruizhi2001/branch-A-Streams

Add use of streams in project.

Ruizhi2001 committed:
Create gradle.yml

Ruizhi2001 committed:
Merge pull request #4 from Ruizhi2001/branch-A-CI

Create gradle.yml

Ruizhi2001 committed:
Add ability to insert locations for tasks

Edit tests to fit new feature

Ruizhi2001 committed:
Merge pull request #5 from Ruizhi2001/branch-D-Places

Add ability to insert locations for tasks

Ruizhi2001 committed:
Update gradle.yml

Ruizhi2001 committed:
Update gradle.yml

Ruizhi2001 committed:
Merge branch 'master' of https://github.com/Ruizhi2001/ip

# Conflicts:
#	.github/workflows/gradle.yml

Ruizhi2001 committed:
Add background and change display picture of DaDuke and DaUser

Ruizhi2001 committed:
A-UserGuide

Update User Guide to be more informative.

Adds the list of commands available and how to use them

Ruizhi2001 committed:
Merge pull request #7 from Ruizhi2001/A-UserGuide

A-UserGuide

Ruizhi2001 committed:
Add Ui png and fix minor parser bug

Ruizhi2001 committed:
Merge pull request #8 from Ruizhi2001/branch-AddUIpng

Add Ui png and fix minor parser bug

Ruizhi2001 committed:
fix parser bugs

Ruizhi2001 committed:
Merge branch 'A-branch-MinorBugFix'

* A-branch-MinorBugFix:
  fix parser bugs

Ruizhi2001 committed:
Fix style issues

Ruizhi2001 committed:
Merge branch 'A-branch-MinorBugFix'

* A-branch-MinorBugFix:
  Fix style issues

Ruizhi2001 committed:
Fix GUI such that list does not cut off with "..."

Ruizhi2001 committed:
Merge branch 'branch-A-MinorBugFix'

* branch-A-MinorBugFix:
  Fix GUI such that list does not cut off with "..."

Ruizhi2001 committed:
Change GUI and update Ui.png

Ruizhi2001 committed:
Merge branch 'branch-A-minorGuiUpdate'

* branch-A-minorGuiUpdate:
  Change GUI and update Ui.png

Ruizhi2001 committed:
Fix minor stylistic errors

Ruizhi2001 committed:
Merge branch 'branch-A-minorGuiUpdate'

* branch-A-minorGuiUpdate:
  Fix minor stylistic errors
